include/deal.II/lac/trilinos_sparsity_pattern.h

```diff
@@ -1376,7 +1382,18 @@ namespace TrilinosWrappers
   inline SparsityPattern::const_iterator
   SparsityPattern::begin() const
   {
-    return const_iterator(this, 0, 0);
+    size_type first_valid_row = 0;
+    while (first_valid_row < this->n_rows())
+      {
+        const auto row_length = this->row_length(first_valid_row);
+        if (row_length == static_cast<SparsityPattern::size_type>(-1))
```

Comment:
same here -- couldn't we find the set of locally owned rows and get its first element?